lib/utils/disk.go

```diff
@@ -37,3 +43,39 @@ func PercentUsed(path string) (float64, error) {
 	ratio := float64(stat.Blocks-stat.Bfree) / float64(stat.Blocks)
 	return Round(ratio * 100), nil
 }
+
+// CanUserWriteTo attempts to check if a user has write access to certain path.
+// It also works around the program being run as root and tries to check
+// the permissions of the user who executed the program as root.
+// This should only be used for string formatting or inconsequential use cases
+// as it's not bullet proof and can report wrong results.
+func CanUserWriteTo(path string) bool {
+	// prevent infinite loops with a maxIterations
+	maxIterations := 10
+	for {
+		if maxIterations == 0 {
+			return false
+		}
+		maxIterations--
+
+		_, err := os.Stat(path)
+		if err == nil {
+			break
+		}
+		if errors.Is(err, fs.ErrNotExist) {
+			path = filepath.Dir(path)
+			continue
+		}
+		return false
+	}
+
+	ogUser := os.Getenv("SUDO_USER")
+	isRoot := ogUser != ""
+
+	if !isRoot {
+		return unix.Access(path, unix.W_OK) == nil
+	}
+
+	cmd := exec.Command("sudo", "-u", ogUser, "test", "-w", path)
```

Comment:
I'm not really a fan of shelling out to external commands (unless necessary). I know it's more work but I wonder what you think about doing it the following way:

- Make a `teleport` helper subcommand e.g. `teleport test-path` or something that implements OS-specific path test. For windows it can be no-op for now, for unix we can use `unix.Access` like you do here.
- Here invoke self (via os.Executable) so effectively `sudo teleport test-path`.

Then we don't rely on any shell commands and have a clean way to extend it to Windows for example in future.

WDYT?